why-choice/index.qmd

```diff
@@ -0,0 +1,265 @@
+    requirements of the Data Package standard into Python ourselves,
+    which would be both error prone and subject to falling out of sync
+    with the standard if it's updated.
+-   We already have a set of classes representing the Data Package
+    standard. The aim of these is to allow users to construct metadata
+    more easily via Sprout. Ideally, our new tool will not require us to
+    develop a parallel set of classes, as we would like to avoid having
+    multiple (potentially inconsistent) internal representations of the
+    standard.
+-   As we treat package, resource and table metadata separately, it
+    would be helpful if we could check these separately against the
+    standard as well.
+-   Another relevant factor is whether we want to allow blank values in
+    our JSON metadata. On the one hand, it may be helpful for users to
+    see a sort of template JSON file with the most important properties
+    present but empty (e.g. set to the empty string or an empty list)
+    when a data package is first set up. On the other hand, such a JSON
+    file does not, technically speaking, conform to the JSON schema of
+    the standard, as the standard disallows empty lists, for example. As
+    different tools come with different default behaviours, different
+    amounts of customisation work are needed depending on our treatment
+    of blank values.
+-   Finally, we would like the mechanism underlying the process of
+    checking metadata against the standard to be transparent, reliable,
+    consistent, and as simple as possible.
+
+## Considered options
+
+::: content-hidden
+List and describe some of the options, as well as some of the benefits
+and drawbacks for each option.
+:::
+
+We consider the following options. There exist alternatives for both
+Pydantic and `jsonschema`, but the pros and cons of these are largely
+similar for the purpose of this comparison. As Pydantic and `jsonschema`
+are the most popular and best supported tools in their categories, only
+these are discussed.
+
+-   [`frictionless-py`](https://framework.frictionlessdata.io/)
+-   [Pydantic](https://docs.pydantic.dev/latest/)
+-   [`jsonschema`](https://python-jsonschema.readthedocs.io/en/stable/)
+
+### frictionless-py
+
+[`frictionless-py`](https://framework.frictionlessdata.io/) is the
+Python implementation of the Data Package standard by its parent
+organisation, and as such it would be the obvious choice for our use
+case. As well as structuring metadata, it has functionality to extract
+metadata from data, run checks on data and metadata, and transform data.
+A variety of sources is supported for both data and metadata.
+
+::::: columns
+::: column
+#### Benefits
+
+-   Comes with a mechanism for checking metadata against the Data
+    Package standard out of the box.
+-   Includes requirements from both the JSON schema and the text-based
+    description of the Data Package standard.
+-   Can check metadata and data separately.
```

Comment:
Maybe this should be qualified a bit in that you can only check data against a schema that conforms to the standard. So you first must check the metadata in order to be able to check the data. And this is indeed how it works in frictionless.

Reply:
Yeah, maybe add that here?